jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1488,13 +1491,21 @@ Method getMethodHelper(
 			return null;
 		}
 	}
-	result = forDeclaredMethod ? getDeclaredMethodImpl(name, parameterTypes, strSig, null) : getMethodImpl(name, parameterTypes, strSig);
+	
+	if (forDeclaredMethod) {
+		result = getDeclaredMethodImpl(name, parameterTypes, strSig, null);
+	} else {
+		result = getMethodImpl(name, parameterTypes, strSig);
+		/* Retrieve the specified method implemented by the superclass from the top to the bottom */
+		resultFromSuperClass = this.isInterface() ? null : getMethodImplBySuperClass(this, name, strSig, parameterTypes);
```

Comment:
This method seems including the result from last call `getMethodImpl(name, parameterTypes, strSig);`, can it be avoided?

Reply:
To avoid the duplication, we should pass in `this.getSuperclass()` instead of `this `(the current class) (Updated).